### ISSUE TITLE
Reconcile support and optimizer-control architecture docs with current main

### ISSUE THREAD

**Issue by Ahmadfauzi34** [Ahmadfauzi34/ai_agent_builder#277]
## Purpose

Contract-coherence cleanup after the repository advanced through native package support, Python host support, structural decoder consolidation, and state-preserving OpenES learning-rate control.

This issue is documentation-only. It should reconcile documents that now lag behind the authoritative/current implementation without redefining runtime semantics.

### 1. Native Rust package support status

`docs/host-support.v1.json` currently records:

```text
native_packages.rust.status = supported
consumer_proof = scripts/audit_rust_package.py
```

but `docs/rust-package-support.md` still says:

```text
Status: candidate support contract pending CI proof
```

and its continuation section still instructs contributors not to begin Python implementation until the Rust package audit passes.

That sequence has already completed: packaged Rust support is proven, ABI v1 exists, and Python is a supported first-class host on the narrow verified matrix.

### 2. Host orchestration versus optimizer control

`docs/es-graph-host-orchestration.md` correctly establishes `KEEP_HOST_ORCHESTRATION`, but predates merged #275.

The document should explicitly record that state-preserving `EsOptimizer.set_learning_rate(...)` is a **core capability** whose invocation policy remains host-owned:

```text
core owns mutation validity + state continuity
host owns whether/when to request the mutation
```

This must not be described as automatic scheduling or as moving objective/controller policy into the core.

## Scope

Update only:

- `docs/rust-package-support.md`
- `docs/es-graph-host-orchestration.md`

No runtime Rust, WASM, ABI, Python, Graph, Math Program, ProgramBundle, Resolution/Authorization, optimizer algorithm, or support-matrix change.

## Required consistency

After the patch, the docs must preserve:

- public Rust API != stable cross-language ABI;
- Rust native package = supported;
- Python first-class host = supported only on the verified matrix;
- ABI v1 remains versioned/experimental, not a long-term stable-ABI promise;
- objective/reward/schedule policy remains host-owned;
- `set_learning_rate` preserves optimizer state and is only a capability, not an embedded scheduling policy;
- program identity, binding identity, mutable learned state, ProgramBundle bytes, and Authorization/Proof remain separate.

## Proof

Documentation-only PR.

Require:

- Rust AI CI and Python Wheel Proof green if scheduled;
- no production files changed;
- exact two-file diff;
- no unresolved review threads.
